src/services/exportInfoMap.ts

```diff
@@ -382,7 +382,12 @@ namespace ts {
         const compilerOptions = program.getCompilerOptions();
         let moduleCount = 0;
         forEachExternalModuleToImportFrom(program, host, /*useAutoImportProvider*/ true, (moduleSymbol, moduleFile, program, isFromPackageJson) => {
-            if (++moduleCount % 100 === 0) cancellationToken?.throwIfCancellationRequested();
+            if (++moduleCount % 100 === 0) {

```

Comment:
passing cleanup callback might not be enough right? Cancellation can be done even in checker and dont you need to protect cache against that. 
May be have cache aware of checker used? or do in finally block of try catch ?

Reply:
Ah yeah, good point. I don’t think the checker actually cancels in practice with these calls, because otherwise I wouldn’t have had to check for cancellation out here. But theoretically you’re right, this could miss stuff.